### PR TITLE
Fix detection and error handling of higher-order inhomogeneous ODEs

### PR DESCRIPTION
**Pull request by clinssen**
The detection of higher-order inhomogeneous ODEs introduced in #51 was too restrictive. The detection has been improved by looking at strongly connected components in the system matrix. A unit test has been added to check this behaviour.

### CODE REVIEW

**Comment by clinssen** [comment]
Note to future reviewers (including myself): never let me get away with raising or catching generic Exceptions, always subclass them into something meaningful (like a PropagatorGenerationException).

File: tests/test_inhomogeneous.py [hunk 1]
```diff
@@ -133,14 +149,45 @@ def test_inhomogeneous_solver_second_order_system(self):
...
         x0 = 0.
         x0d = 10.
 
-        shape = Shape.from_ode("x", "-1/tau**2 * x - 2/tau * x' - 1", initial_values={"x": str(x0), "x'": str(x0d)}, parameters=parameters_dict)
-        sys_of_shape = SystemOfShapes.from_shapes([shape], parameters=parameters_dict)
+        shape_V_m = Shape.from_ode("V_m", "x / tau + (E_L - V_m)", initial_values={"V_m": "0."}, parameters=parameters_dict)
+        shape_I_syn1 = Shape.from_ode("x", "y", initial_values={"x": str(x0)}, parameters=parameters_dict)
+        shape_I_syn2 = Shape.from_ode("y", "-1/tau**2 * x - 2/tau * y", initial_values={"y": str(x0d)}, parameters=parameters_dict)
+        sys_of_shape = SystemOfShapes.from_shapes([shape_V_m, shape_I_syn1, shape_I_syn2], parameters=parameters_dict)
         solver_dict = sys_of_shape.generate_propagator_solver()
```
Comment (pnbabu) [inline tests/test_inhomogeneous.py:178]: Is it possible to add assertions to this test?

**Review by pnbabu — COMMENTED** [review]
Looks fine to me. I just have a small comment.

File: tests/test_inhomogeneous.py (hunk 1 above)
Comment (clinssen) [reply tests/test_inhomogeneous.py:178]: Done!

**Review by pnbabu — APPROVED** [review]

**Comment by clinssen** [comment]
Thanks for the review!

**Pull request merged**